src/analysis/normed_space/basic.lean

```diff
@@ -569,6 +569,33 @@ instance submodule.normed_space {𝕜 : Type*} [normed_field 𝕜]
 
 end normed_space
 
+section normed_algebra
+
+/-- A normed algebra `𝕜'` over `𝕜` is an algebra endowed with a norm for which the embedding of
+`𝕜` in `𝕜'` is an isometry. -/
+class normed_algebra (𝕜 : Type*) (𝕜' : Type*) [normed_field 𝕜] [normed_ring 𝕜']
+  extends algebra 𝕜 𝕜' :=
+(norm_eq : ∀x:𝕜, ∥algebra_map 𝕜' x∥ = ∥x∥)
+
```

Comment:
Should there be a lemma `isometry_algebra_map`, that restates the `norm_eq` condition as `isometry (algebra_map _ : 𝕜 → 𝕜')?

Also, should `normed_algebra.norm_eq` be turned into a `simp`-lemma?

Reply:
Done and done.